freqtrade/commands/data_commands.py

```diff
@@ -160,7 +160,6 @@ def start_list_data(args: Dict[str, Any]) -> None:
     from freqtrade.data.history.idatahandler import get_datahandler
     dhc = get_datahandler(config['datadir'], config['dataformat_ohlcv'])
 
-    # TODO-lev: trading-mode should be parsed at config level, and available as Enum in the config.
```

Comment:
I don't think we should have a is_short_strategy flag that's at the config, I think that just the presence of enter_short exit_short should be enough, and we can through an exception if it's spot.

It would be a good idea to be able to turn off shorts or longs entirely based on the market though (if the market goes bull, turn off short, if it goes bear turn off long), but that wouldn't be in the config, it would be dynamic within the strategy

Reply:
>  we can through an exception if it's spot.

Please start thinking about things from a user perspective ("how would that feel as a user").

If you throw an exception once a signal appears - you'd crash the bot potentially days into running the bot - which from a UX - is VERY bad.

if you run spot there's 2 ways - either we ignore `enter_short` completely (no warning, just not done) - or we have a flag similar to `use_custom_stoploss` (for example) - which acts as "optin" for shorts - and blocks execution on spot markets (so you're aware that you're running a short strategy on spot markets).

Throwing an exeption if a signal on the wrong market appears is out of the question.

Reply:
That said - the above TODO was about something else - which is NOT done and should therefore be kept there.